lightbug_http/server.mojo

```diff
@@ -132,7 +132,7 @@ struct Server:
         _ = self.set_address(address)
         self.serve(listener, handler)
 
-    fn serve[T: HTTPService](inout self, ln: NoTLSListener, handler: T) raises -> None:
+    fn serve[T: HTTPService](inout self, ln: NoTLSListener, inout handler: T) raises -> None:
```

Comment:
I guess this will be more relevant later once we introduce async, but I wonder if always having a handler passed as mutable is potentially dangerous in concurrent applications

Reply:
I agree we will likely have to make changes once Mojos async story is more complete, but maybe by then we'll have a better mechanism for this (maybe something like the `mutable` keyword in C++?)